Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -0,0 +1,333 @@
+                Flag.IS_PUBLIC,
+                Flag.IS_FINAL,
+                Flag.PropertyAccessor.IS_EXTERNAL
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val PropertyStub.setterFlags: Flags
+        get() = listOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+                Flag.IS_PUBLIC,
+                Flag.IS_FINAL,
+                Flag.PropertyAccessor.IS_EXTERNAL
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val StubType.flags: Flags
+        get() = listOfNotNull(
+                Flag.Type.IS_NULLABLE.takeIf { nullable }
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val TypealiasStub.flags: Flags
+        get() = listOfNotNull(
+                Flag.IS_PUBLIC
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    fun AnnotationStub.map(): KmAnnotation {
+        val args = when (this) {
+            AnnotationStub.ObjC.ConsumesReceiver -> TODO()
+            AnnotationStub.ObjC.ReturnsRetained -> TODO()
+            is AnnotationStub.ObjC.Method -> TODO()
+            is AnnotationStub.ObjC.Factory -> TODO()
+            AnnotationStub.ObjC.Consumed -> TODO()
+            is AnnotationStub.ObjC.Constructor -> TODO()
+            is AnnotationStub.ObjC.ExternalClass -> TODO()
+            AnnotationStub.CCall.CString -> TODO()
+            AnnotationStub.CCall.WCString -> TODO()
+            is AnnotationStub.CCall.Symbol ->
+                mapOf("id" to KmAnnotationArgument.StringValue(symbolName))
+            is AnnotationStub.CStruct -> TODO()
+            is AnnotationStub.CNaturalStruct -> TODO()
+            is AnnotationStub.CLength -> TODO()
+            is AnnotationStub.Deprecated -> TODO()
+        }
+        return KmAnnotation(classifier.fqNameSerialized, args)
+    }
+
+    fun StubType.map(): KmType {
+        return when (this) {
+            is ClassifierStubType -> KmType(flags).also { km ->
+                km.arguments += typeArguments.map { it.map() }
+                if (isTypealias) {
+                    km.abbreviatedType = abbreviatedType
+                    km.classifier = expandedType.map().classifier
+                } else {
+                    km.classifier = KmClassifier.Class(classifier.fqNameSerialized)
+                }
+
+            }
+            is FunctionalType -> KmType(flags).also { km ->
+                km.classifier = KmClassifier.Class(classifier.fqNameSerialized)
+            }
+            is TypeParameterType -> KmType(flags).also { km ->
+                km.classifier = KmClassifier.TypeParameter(id)
+            }
+        }
+    }
+
+    private val ClassifierStubType.abbreviatedType: KmType
+        get() = KmType(flags).also { km ->
+            km.classifier = KmClassifier.TypeAlias(classifier.fqNameSerialized)
+        }
+
+    fun FunctionParameterStub.map(): KmValueParameter =
+            KmValueParameter(flags, name).also { km ->
+                type.map().let {
```

Comment:
I find this usage of `.let` somewhat redundant.
Storing `type.map()` to a variable could make the code simpler.